hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientRingbufferProxy.java

```diff
@@ -243,8 +246,20 @@ private static void checkSequence(long sequence) {
         try {
             final Future future = new ClientInvocation(getClient(), clientMessage, partitionId).invoke();
             return (T) future.get();
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof StaleSequenceException) {
+                StaleSequenceException se = (StaleSequenceException) e.getCause();
+                long l = headSequence();
+                throw new StaleSequenceException(se.getMessage(), l);
```

Comment:
Can't the original exception be set as cause? So you get a bit if a funky StaleSequenceException caused by StaleSequenceException.